src/provider/persistence/dataLayer/repositories/Repository.ts

```diff
@@ -1,8 +1,76 @@
-import {Entity} from '../../../../client/Entity';
-import {ReturnResult, VoidResult} from '../../../model/Result';
 import {PageInfo} from '../../models/PageInfo';
 import {IDatastore} from '../IDatastore';
 
+/**
+ * @description Base model for any stored items
+ */
+export interface Entity {
+    id: string;
+}
+
+/**
+ * Object that returns the status of a database operation.
+ *
+ * This interface is used for operations that return data. If an operation has
+ * no return value, it will use VoidResult instead.
+ */
+export type ReturnResult<T> = ReturnResultSuccess<T>|ReturnResultFailure<T>;
+
+export interface ReturnResultSuccess<T> extends VoidResultSuccess {
+    success: true;
+
+    /**
+     * Returns the results of the operation.
+     *
+     * If the operation failed (see 'success'), null will be returned.
+     */
+    value: T;
+}
+export interface ReturnResultFailure<T> extends VoidResultFailure {
+    success: false;
+
+    /**
+     * Returns the results of the operation.
+     *
+     * If the operation failed (see 'success'), null will be returned.
+     */
+    value: null;
+}
+
+/**
+ * Object that reutrns the status of a database operation.
```

Comment:
'returns'

Reply:
Ah, ok, not your typo

Reply:
I fixed some of those already, guess I missed one.